src/app/base/components/LimitedIpInput/LimitedIpInput.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState } from "react";
+
+import { Input } from "@canonical/react-components";
+import type { InputProps } from "@canonical/react-components";
+import { useFormikContext } from "formik";
+
+import type { Subnet } from "@/app/store/subnet/types";
+import {
+  getImmutableAndEditableOctets,
+  getIpRangeFromCidr,
+} from "@/app/utils/subnetIpRange";
+
+type Props = Omit<InputProps, "maxLength" | "placeholder" | "type" | "name"> & {
+  cidr: Subnet["cidr"];
+  name: string;
+};
+
+const LimitedIpInput = ({ cidr, name, ...props }: Props) => {
```

Comment:
I've not heard of that, this could definitely be useful! My only concern is that it was last updated 4 years ago, would be nice if there was something similar that's still in active development